plugins/default/src/Ide/Plugin/Eval.hs

```diff
@@ -280,38 +271,71 @@ runEvalCmd lsp state EvalParams {..} = withIndefiniteProgress lsp "Eval" Cancell
 
     return (WorkspaceApplyEdit, ApplyWorkspaceEditParams workspaceEdits)
 
+
+type GHCiLikeCmd = DynFlags -> Text -> Ghc (Maybe Text)
+
+-- Should we use some sort of trie here?
+ghciLikeCommands :: [(Text, GHCiLikeCmd)]
+ghciLikeCommands =
+  [ ("kind", doKindCmd False)
+  , ("kind!", doKindCmd True)
+  , ("type", doTypeCmd)
+  ]
+
 evalGhciLikeCmd :: Text -> Text -> Ghc (Maybe Text)
 evalGhciLikeCmd cmd arg = do
   df <- getSessionDynFlags
-  let tppr = T.pack . showSDoc df . ppr
-  case cmd of
-    "kind" -> do
-      let input = T.strip arg
-      (_, kind) <- typeKind False $ T.unpack input
-      pure $ Just $ "-- " <> input <> " :: " <> tppr kind <> "\n"
-    "kind!" -> do
-      let input = T.strip arg
-      (ty, kind) <- typeKind True $ T.unpack input
-      pure
-        $ Just
-        $ T.unlines
-        $ map ("-- " <>)
-        [ input <> " :: " <> tppr kind
-        , "= " <> tppr ty
-        ]
-    "type" -> do
-      let (emod, expr) = parseExprMode arg
-      ty <- exprType emod $ T.unpack expr
-      pure $ Just $
-        "-- " <> expr <> " :: " <> tppr ty <> "\n"
-    _ -> E.throw $ GhciLikeCmdNotImplemented cmd arg
+  case lookup cmd ghciLikeCommands
+    <|> snd <$> find ((T.isPrefixOf cmd).fst) ghciLikeCommands of
```

Comment:
I think `(T.isPrefixOf cmd . fst)` is better, as
1. You don't need parentheses for function application inside of function composition
2. function composition (`.`) should have spaces around it so that it will be compatible with https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0282-record-dot-syntax.rst

Reply:
Agh, the parens are accidentally introduced because I had once written as ``(cmd `T.isPrefixOf`)`` and realised this doesn't require infixes at all but forgot to drop the parens.

> function composition (.) should have spaces around it so that it will be compatible with https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0282-record-dot-syntax.rst

Yes, I forgot that the RecordDotSyntax is nearly coming. I will fix that shortly!